### ISSUE TITLE
Add support for different references

### ISSUE THREAD

**Issue by stevengoossensB** [elastic/elastic-package#2176]
I would like to have the possibility to add own references in the build.yml for fields that are not in ecs, but standardized across my set of log sources. Those fields will be defined in the same yaml structure, but need to have a different name.

The fact that this needs to be **ecs** seems to be hardcoded at 2 different places.

![Image](https://github.com/user-attachments/assets/9ac02ce8-d05c-489d-9257-afe7118a805d)
![Image](https://github.com/user-attachments/assets/872afe37-bff7-433f-b089-fa73a2cdbc84)
![Image](https://github.com/user-attachments/assets/b1759552-1f01-4264-b36a-86509353842e)

**Comment by jsoriano** [comment]
Yes, indeed this feature only supports ecs so far. We would need to define a common struct for other repositories. I will move the issue to the package-spec repository.
